examples/vlan_learning.c

```diff
@@ -31,6 +36,13 @@ int handle_phys2virt(struct __sk_buff *skb) {
       // relearn when mac moves ifindex
       if (out_leaf->out_ifindex != skb->ifindex)
         out_leaf->out_ifindex = skb->ifindex;
+      // relearn when vlan_tci/vlan_proto changes
+      if (out_leaf->vlan_tci != skb->vlan_tci)
+        out_leaf->vlan_tci = skb->vlan_tci;
+      if (out_leaf->vlan_proto != skb->vlan_proto)
+        out_leaf->vlan_proto = skb->vlan_proto;
+      // pop the vlan header and send to the destination
+      bpf_skb_vlan_pop(skb);
```

Comment:
In my old python changes, I disabled ipv6 in various places so that tcpdump
packets will be clean.
However, looks like pyroute2 does not really work well with "sysctl ..."
(This happens before with confusing messages), and
the worker http server (in worker namespace: python -m SimpleHTTPServer 80)
does not come up, thus tcp connection cannot be
established.

Without disabling ipv6, everything works fine.

I tried to use "ip" command to create veth, create namespace and disable
ipv6 inside namespace and launch SimpleHTTPServer
inside the namespace, there is no issue.


On Wed, Jul 22, 2015 at 10:48 AM, 4ast <notifications@github.com> wrote:

> In examples/vlan_learning.c
> <https://github.com/iovisor/bcc/pull/90#discussion_r35241879>:
>
> > @@ -31,6 +36,13 @@ int handle_phys2virt(struct __sk_buff *skb) {
> >        // relearn when mac moves ifindex
> >        if (out_leaf->out_ifindex != skb->ifindex)
> >          out_leaf->out_ifindex = skb->ifindex;
> > +      // relearn when vlan_tci/vlan_proto changes
> > +      if (out_leaf->vlan_tci != skb->vlan_tci)
> > +        out_leaf->vlan_tci = skb->vlan_tci;
> > +      if (out_leaf->vlan_proto != skb->vlan_proto)
> > +        out_leaf->vlan_proto = skb->vlan_proto;
> > +      // pop the vlan header and send to the destination
> > +      bpf_skb_vlan_pop(skb);
>
> looks great. so what was the problem with rst+ack ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iovisor/bcc/pull/90/files#r35241879>.
>